software/contrib/poly_square.py

```diff
@@ -0,0 +1,246 @@
+            PolyphonyMode("Major 7th", (0, 4/12, 11/12)),
+            PolyphonyMode("Minor 7th", (0, 3/12, 10/12))
+        ]
+        self.current_mode = None
+        self.ui_update_requested = True
+        self.detune_amount = None
+        self.coarse_tune = .5
+        self.fine_tune = .5
+        self.tuning_mode = False
+        self.load_state()
+
+        @b1.handler
+        def tuning_mode_on():
+            self.tuning_mode = True
+            self.ui_update_requested = True
+
+        @b1.handler_falling
+        def tuning_mode_off():
+            self.tuning_mode = False
+            self.ui_update_requested = True
+            # Saves the tuning settings when the button is released
+            self.save_state()
+        
+    # Converts V/oct signal to Hertz, with 0V = C0
+    def get_hertz(self, voltage):
+        # Start with A0 because it's a nice, rational number
+        a0 = 27.5
+        # Subtract 3/4 from the voltage value so that 0V = C0
+        return a0 * 2 ** (voltage - 3/4)
+    
+    # Returns the linear step distance between elements such that all are
+    # equally spaced. Ex, for 6 elements between 0 and 100 (inclusive), the
+    # step would be 20 (the elements would then be 0, 20, 40, 60, 80, 100).
+    def get_step_distance(self, first, last, count):
+        return (last - first) / (count - 1)
+    
+    # Returns the total voltage offset of the current tuning values
+    def get_tuning(self):
+        return self.coarse_tune - .5 + (self.fine_tune - .5) / 12
+    
+    # Through-zero detuning of current oscillator based on its position in
+    # the list of oscillators - the central oscillator (assuming an odd 
+    # number of oscillators) stays in tune while the outer oscillators are 
+    # progressively detuned.
+    def get_detuning(self, detune_amount, oscillator):
+        step_distance = self.get_step_distance(0, 1, len(self.oscillators))
+        return detune_amount * (step_distance *
+            self.oscillators.index(oscillator) - step_distance)
+    
+    # Returns a voltage offset for the oscillator based on the current
+    # polyphony mode. This allows for things like triads, with the offsets
+    # set accordingly.
+    def get_offset(self, oscillator):
+        return self.modes[self.current_mode].voltage_offsets[
+            self.oscillators.index(oscillator)]
+    
+    # Saves oscillator tuning settings
+    def save_state(self):
+        settings = {"c": self.coarse_tune, "f": self.fine_tune}
+        self.save_state_json(settings)
+
+    # Loads oscillator tuning settings
+    def load_state(self):
+        settings = self.load_state_json()
+        if "c" in settings:
+            self.coarse_tune = settings["c"]
+        if "f" in settings:
+            self.fine_tune = settings["f"]
+    
+    def update_ui(self):
+        if self.tuning_mode:
+            text = ("Tuning Mode\ncoarse: " + str(self.coarse_tune - .5) +
+                "\nfine: " + str(self.fine_tune - .5))
```

Comment:
No, not really - I mostly included that for some visual feedback. I think I'll probably push a change which will give you bars that fill up instead of numbers, since the numbers aren't really meaningful to the user.

Reply:
That sounds like a nice UI. There's another script that does something similar and it works well there. 

Don't feel like you need to make this change in this PR if you don't want to take then time right now.